ppp: compile fix: unset FILTER variable in Makefile

rittere opened the pull request:
This patch fixes a compilation problem within ppp. Compiling the sources produces

Package ppp is missing dependencies for the following libraries:
libpcap.so.1
make[3]: *** [Makefile:322: /usr/local/src/embedded/openwrt/openwrt_github/bin/packages/mips_24kc/base/ppp_2.4.9.git-2021-01-04-3_mips_24kc.ipk] Error 1
make[3]: Leaving directory '/usr/local/src/embedded/openwrt/openwrt_github/package/network/services/ppp'
time: package/network/services/ppp/default/compile#0.24#0.07#0.32
    ERROR: package/network/services/ppp failed to build (build variant: default).
make[2]: *** [package/Makefile:114: package/network/services/ppp/compile] Error 1

The fix consists of moving the dependency of ppp on libcap to the packages Default and multilink (and hence to all packages).

Signed-off-by: Eike Ritter <git@rittere.co.uk>

adschm commented:
This has formal issues, please have another look at https://openwrt.org/submitting-patches

And please don't close and open a new PR, but update this one.

rittere commented:
I've updated the commit - message.

nbd168 commented:
Not sure what changed here, but we used to link the static libpcap library to avoid this runtime dependency, which adds a sizable chunk of bloat to the default images. I think we should continue to do so.

rittere committed:
ppp: compile fix: unset FILTER variable in Makefile

If the environment variable FILTER is set before compilation,
compilation of the ppp-package will fail with the error message

Package ppp is missing dependencies for the following libraries:
libpcap.so.1

The reason is that the OpenWrt-patch for the Makefile only comments
out the line FILTER=y. Hence the pcap-library will be dynamically
linked if the environment variable FILTER is set elsewhere, which
causes compilation to fail. The fix consists on explicitly unsetting
the variable FILTER instead.

Signed-off-by: Eike Ritter <git@rittere.co.uk>

rittere commented:
In view of nbd168's comments I had a look at my compilation problems again. It turns out that the cause is the environment variable FILTER. My environment sets this to some value. However, the pppd-Makefile assumes that it is not set. If it is set, the pcap-library will be dynamically linked, which causes compilation to fail, as described above. This can be fixed by using FILTER=  instead of commenting out FILTER=y in the Makefile for pppd. At the moment the pcap-library  is always statically linked, and this option is set in the Makefile, hence it should be OK to always unset the option of linking the pcap-library dynamically.

hauke commented:
Thank you for your pull request, I applied to to master.
